platform/server/cpp/src/logger.rs

```diff
@@ -0,0 +1,355 @@
+use bd_session::{Store, Strategy};
+use std::collections::HashMap;
+use std::hash::{DefaultHasher, Hash, Hasher};
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::time::Instant;
+use tokio::try_join;
+
+//
+// RustLogger
+//
+
+pub struct RustLogger {
+  _logger: bd_logger::Logger,
+  handle: bd_logger::LoggerHandle,
+  session_strategy: Arc<Strategy>,
+  device: Arc<bd_logger::Device>,
+}
+
+impl RustLogger {
+  pub fn new(
+    api_key: String,
+    api_address: &str,
+    sdk_directory: String,
+    app_id: String,
+    app_version: String,
+    os: String,
+    os_version: String,
+    locale: String,
+  ) -> anyhow::Result<Self> {
+    let start = Instant::now();
+    let sdk_directory = PathBuf::from(sdk_directory);
+
+    let storage = Box::new(DiskStorage::new(sdk_directory.join("storage"))?);
+    let store = Arc::new(Store::new(storage));
+    let device = Arc::new(bd_device::Device::new(store.clone()));
+    let device_clone = device.clone();
+    let session_strategy = Arc::new(Strategy::Fixed(fixed::Strategy::new(
+      store.clone(),
+      Arc::new(UUIDCallbacks),
+    )));
+    let session_strategy_clone = session_strategy.clone();
+    let shutdown = bd_shutdown::ComponentShutdownTrigger::default();
+
+    let metadata_provider = Arc::new(MetadataProvider::new(
+      app_id.clone(),
+      app_version.clone(),
+      os,
+      os_version,
+      locale,
+    ));
+
+    let static_metadata = Arc::new(StaticMetadata::new(
+      app_id,
+      app_version,
+      // TODO(fz): We should add Platform::Server
+      Platform::Electron,
+      device.id(),
+    ));
+
+    let (network, handle) =
+      bd_hyper_network::HyperNetwork::new(api_address, shutdown.make_shutdown());
+
+    let reporter = {
+      let (reporter, handle) =
+        bd_hyper_network::ErrorReporter::new(api_address.to_string(), api_key.clone());
+
+      let handle = bd_client_common::error::MetadataErrorReporter::new(
+        Arc::new(handle),
+        Arc::new(SessionProvider {
+          strategy: session_strategy.clone(),
+        }),
+        static_metadata.clone(),
+      );
+
+      bd_client_common::error::UnexpectedErrorHandler::set_reporter(Arc::new(handle));
+
+      reporter
+    };
+
+    let (logger, _, logger_future) = LoggerBuilder::new(InitParams {
+      sdk_directory,
+      api_key,
+      session_strategy: session_strategy_clone,
+      store,
+      metadata_provider,
+      resource_utilization_target: Box::new(EmptyTarget),
+      session_replay_target: Box::new(EmptyTarget),
+      events_listener_target: Box::new(EmptyTarget),
+      device: device_clone,
+      network: Box::new(handle),
+      static_metadata,
+    })
+    .with_client_stats(true)
+    .build()?;
+
+    LoggerBuilder::run_logger_runtime(async {
+      // Make sure we hold onto the shutdown handle to avoid an immediate shutdown.
+      #[allow(clippy::no_effect_underscore_binding)]
+      let _shutdown = shutdown;
```

Comment:
nit: to avoid the clippy lint usually I just `drop(shutdown)` at the end of the block.